pkg/apis/environment/extension.go

```diff
@@ -353,3 +358,131 @@ func (h Handler) RouteStop(req RouteStopRequest) error {
 }
+
+func (h Handler) RouteApply(req RouteApplyRequest) error {
+	sj := session.MustGetSubject(req.Context)
+
+	token, err := auths.GetAccessToken(
+		req.Context, h.modelClient,
+		sj.ID, types.TokenKindAPI, types.WalrusOperationTokenName)
+	if err != nil {
+		if model.IsNotFound(err) {
+			token, err = auths.CreateAccessToken(req.Context, h.modelClient,
+				sj.ID, types.TokenKindAPI, types.WalrusOperationTokenName, nil)
+			if err != nil {
+				return errorx.New(err.Error())
+			}
+		}
+	}
+
+	sc := serverContext(req.Project.Name, req.Name, token.AccessToken)
+
+	loader := manifest.DefaultLoader(sc, true)
+
+	set, err := loader.LoadFromByte([]byte(req.YAML))
+	if err != nil {
+		return errorx.Wrap(err, "failed to load walrus file")
+	}
+
+	operator := manifest.DefaultApplyOperator(sc, false)
+
+	r, err := operator.Operate(set)
+	if err != nil {
+		return errorx.Wrap(err, "failed to apply walrus file")
+	}
+
+	if r.Failed.Len() != 0 {
+		var keys []string
+		for _, v := range r.Failed.All() {
+			keys = append(keys, v.Key())
+		}
+
+		return errorx.Errorf("failed to apply: %v", keys)
+	}
+
+	return nil
+}
+
+func (h Handler) RouteExport(req RouteExportRequest) error {
+	res, err := h.modelClient.Resources().Query().
+		Where(
+			resource.ProjectID(req.Project.ID),
+			resource.EnvironmentID(req.EnvironmentQueryInput.ID),
+			resource.IDIn(req.IDs...)).
+		Select(
+			resource.FieldName,
+			resource.FieldDescription,
+			resource.FieldLabels,
+			resource.FieldAttributes,
+			resource.FieldType,
+			resource.FieldTemplateID).
+		WithTemplate(func(tvq *model.TemplateVersionQuery) {
+			tvq.Select(
+				templateversion.FieldName,
+				templateversion.FieldVersion)
+		}).
+		All(req.Context)
+	if err != nil {
+		return err
+	}
+
+	resSpec := make([]types.ResourceSpec, len(res))
+	for i := range res {
+		// Resources.
+		resSpec[i] = toResourceSpec(req, res[i])
+	}
+
+	// Export.
+	wf := &types.WalrusFile{
+		Version:   types.WalrusFileVersion,
+		Resources: resSpec,
+	}
+
+	yml, err := wf.Yaml()
+	if err != nil {
+		return errorx.Wrap(err, "failed to marshal walrus file")
+	}
+
+	name := filename(req, resSpec)
+	req.Context.Writer.WriteHeader(200)
+	req.Context.Writer.Header().Set("Content-Type", "text/yaml")
+	req.Context.Writer.Header().
+		Set("Content-Disposition", fmt.Sprintf(`attachment; filename="%s"`, name))
+	_, _ = req.Context.Writer.Write(yml)
+
+	return nil
+}
+
+func filename(req RouteExportRequest, resSpec []types.ResourceSpec) string {
+	name := fmt.Sprintf("%s-%s.yaml", req.Project.Name, req.EnvironmentQueryInput.Name)
+	if len(resSpec) == 1 {
+		name = fmt.Sprintf("%s-%s-%s.yaml", req.Project.Name, req.EnvironmentQueryInput.Name, resSpec[0].Name)
+	}
+
+	return name
+}
+
+func toResourceSpec(req RouteExportRequest, res *model.Resource) types.ResourceSpec {
+	spec := types.ResourceSpec{
+		Project: &types.MetadataName{
+			Name: req.Project.Name,
+		},
+		Environment: &types.MetadataName{
+			Name: req.EnvironmentQueryInput.Name,
```

Comment:
What happen if we export a resource from env A then apply to env B?

Reply:
While the exported file(A) includes the scope(project/environment), and the user applying with this file(Apply to B) will use the scope in the file. 
While the exported file doesn't include scope, apply will use the scope from API path params. To make the exported file apply/create to another env, prefer not to include the scope in the file.